### DISCUSSION TITLE
Custom tag to import code blocks for display

### DISCUSSION THREAD

**Discussion by davidyeiser** [markdoc/markdoc#346] [Show and tell]
We used Markdoc and Next.js to build [our (new-ish) docs site](https://www.nextmv.io/docs). We liked the separation of concerns that Markdoc offered (all the content lives in a separate `content` directory) and the ease of flexibility to build custom tags for certain use cases. Overall it’s been a success — we have multiple people writing docs at any given time and it’s been easy for people to add or edit the docs since all they’re interacting with is markdown files.

But what I wanted to highlight was a custom Markdoc tag we created that **takes an external file and renders it as a syntax-highlighted code block**. We did this so when you’re writing docs you don’t have to copy and paste a bunch of code that lives inline with the rest of your content. You can keep the code separated from the content. 

So for example markdown like this:

```
Leave out the lines attribute to display the entire block:

{% ref path="@docs/code/go/sdk/example.go" label="Go" language="go" /%}

Specify a range of lines:

{% ref path="@docs/code/go/sdk/example.go" label="Go" language="go" lines="3-7" /%}

Specify one line:

{% ref path="@docs/code/go/sdk/example.go" label="Go" language="go" lines="2" /%}

Specify scattered lines:

{% ref path="@docs/code/go/sdk/example.go" label="Go" language="go" lines="1-2, 8-9" /%}
```

Would render on the site like this:

<img width="988" alt="blog-docs-site-rendered-code-snippets" src="https://user-images.githubusercontent.com/175191/226909990-ad2b64b4-c2fd-4ff4-87b2-c01d68a1e2fe.png">

If you want to see how we set it up, we have a 
- blog post — [**Building a docs site with Next.js and Markdoc**](https://www.nextmv.io/blog/building-a-docs-site-with-next-js-and-markdoc)
- sample GitHub repository — [**nextmv-io/docs-code-example**](https://github.com/nextmv-io/docs-code-example).

**Comment by mfix-stripe** [comment]
Super neat @davidyeiser! Thanks for sharing 🙏

**Reply by davidyeiser** [reply]
Thanks! And my pleasure!
